### PR TITLE
[WIP] feat: send to only one client if gateway is DB-backed

### LINKED ISSUE #3606: When Gateway discovery is enabled and Kong is DB backed, KIC only sends config to one of Kong gateway endpoint
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

In DB-backed deployment of kong gateway,  KIC should only send to one of gateway admin API endpoints, and kong gateway instances could do the synchronization of configuration via DB. So, instead of limiting gateway discvoery/SCD to be used with only DB-less mode (#3401), we could let KIC to send configuration on any one of available gateway clients when gateway discovery is enabled, to enable KIC to work with multiple DB-backed gateway instances. 

### Proposed Solution

- When gateway discovery is enabled, and kong gateway is DB-backed, KIC send kong configurations to only one of gateway clients.

### Additional information

_No response_

### Acceptance Criteria

- [ ] When kong gateway is DB-backed, and KIC is configured to enable gateway discovery, KIC only sends configuration to one gatewat pod.
- [ ] In the deployment above, all gateway nodes could receive the same configuration.
- [ ] Have an "all-in-one-postgres-multiple-gw.yaml" that could deploy KIC with gateway discovery + kong with postgres DB backend

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

sends to only one of available gateway admin API clients when gateway is DB-backed. When kong gateway is deployed in DB-less mode, configurations should be sent to ALL gateway pods. While in DB-backed deployment, configurations should be sent be only one client.

TODO:
- [x] Add all-in-one manifests for postgres-multi-gw 
- [x] Add ~~integration~~ e2e tests to verify since we could not modify config of KIC in integration tests (depends on an all-in-one manifests)
(The two things above: in #3633 )
- [ ] check if configurations are sent to only one pod of gateways

**Which issue this PR fixes**:

implementation & test part of #3606 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/dataplane/kong_client.go [hunk 1]
```diff
@@ -460,7 +460,23 @@ func (c *KongClient) sendOutToClients(
 	ctx context.Context, s *kongstate.KongState, formatVersion string, config sendconfig.Config,
 ) ([]string, error) {
 	clients := c.clientsProvider.AllClients()
+	if len(clients) == 0 {
+		c.logger.Debugf("no available kong clients, skipping sending out configurations")
+		return c.SHAs, nil
+	}
 	c.logger.Debugf("sending configuration to %d clients", len(clients))
+	// send to only one client if gateway is DB-backed.
```
Comment (czeslavo) [inline internal/dataplane/kong_client.go:468]: We have to take into account that `AllClients` returns the Konnect client if configured. I'd propose changing this to use `GatewayClients()` and adding a `KonnectClient() (*adminapi.Client, bool)` method to handle Konnect separately always.

Before this, the separation wasn't really needed and we could treat all clients (including Konnect) uniformly, but now it changes.

**Review by czeslavo — CHANGES_REQUESTED** [review]

File: internal/dataplane/kong_client.go (hunk 1 above)
Comment (pmalek) [reply internal/dataplane/kong_client.go:468]: I'm pretty sure this was brought up somewhere: do we have to send it to the elected leader or we just pick one, arbitrary instance that will receive the config and we're good?

Can we expect any issues if we send config at point in time N, to instance X, and in N+1 we send it to instance Y?

Comment (czeslavo) [reply internal/dataplane/kong_client.go:468]: According to the [Admin API docs](https://docs.konghq.com/gateway/latest/admin-api/) that @randmonkey has pointed me to, we can call an arbitrary instance:

> Requests to the Admin API can be sent to any node in the cluster, and Kong will keep the configuration consistent across all nodes.

Comment (randmonkey) [reply internal/dataplane/kong_client.go]: Moved to method `sendOutToClientsDBMode` and in the method we send config to ALL konnect nodes(could be 0 or 1 clients in current state), and ONE gateway client.

**Pull request closed without merging**